source/data-formats/working-with-bson.txt

```diff
@@ -14,386 +14,237 @@ Working with BSON
+To create a BSON document by using the list builder, construct a ``bsoncxx::builder::list``
+object, passing a list of key-value pairs to the constructor. The list builder creates a
+BSON document if the key-value list fulfills the following requirements:
 
-      // We're returning a view of the local value
-      return stack_value.view(); // Bad!!
-   }
+- The list has an even number of elements.
+- Each key is a ``string`` type and each value is a ``bson_value::value`` type or is
+  implicitly convertible to one.
 
-This method returns a dangling view that should not be used:
+If the preceding requirements are not met, the list builder creates an array. 
 
-.. code-block:: cpp
+This example uses the list builder to perform the following actions:
 
-   // This view contains a dangling pointer
-   bsoncxx::document::view dangling_view = make_a_dangling_view(); // Warning!!
+- Build a document 
+- Build an array
+- Convert the builder document to a ``bsoncxx::document::value`` object
 
-Attempting to create a view off of a builder will similarly create a
-dangerous view object, because the temporary value returned from
-``extract()`` isn't captured:
+.. literalinclude:: /includes/data-formats/bson.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-bson-list
+    :end-before: end-bson-list
 
-.. code-block:: cpp
+.. _cpp-bson-basic:
 
-   bsoncxx::builder::stream::document temp_builder{};
-   temp_builder << "oh" << "no";
-   bsoncxx::document::view dangling_view = temp_builder.extract().view(); // Bad!!
+Basic Builder
+~~~~~~~~~~~~~
+
+The ``builder::basic`` interface is a builder-style interface for constructing a BSON document. 
+To create a BSON document by using the basic builder, specify your document's data in a list of
+key-value pair objects. You can create these key-value pair objects by passing a key and value
+to the ``builder::basic::kvp()`` method. The key must be a ``string`` type, and
+the value must be a ``bson_value::value`` type or implicitly convertible to one. 
+
+You can use the basic builder's ``make_document()`` method to create a document and
+convert it to a ``bsoncxx::document::value`` in a single statement, as shown in the
+following code:
+
+.. literalinclude:: /includes/data-formats/bson.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-bson-make-document
+    :end-before: end-bson-make-document
+
+Alternatively, you can create a BSON document across multiple statements
+by appending key-value pairs to a basic builder object. This 
+example uses the basic builder to perform the following actions:
+
+- Initialize a ``builder::basic::document`` object
+- Store data in the document by using the ``append()`` method
+- Convert the builder document to a ``bsoncxx::document::value`` object
+
+.. literalinclude:: /includes/data-formats/bson.cpp
+    :language: cpp
+    :dedent:
+    :start-after: start-bson-append
+    :end-before: end-bson-append
+
+.. _cpp-bson-stream:
+
+Stream Builder
```

Comment:
Note to tech reviewer: should we document the stream builder or just remove this section?

Reply:
Sorry, I missed this on first review. I support keeping the documentation for the stream builder with the warning to discourage use. The stream builder is currently [documented](https://github.com/mongodb/mongo-cxx-driver/blob/7402d3d8ffc093fdc647fb2b2469f3a154beb41e/src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/builder/stream/document.hpp#L35) as discouraged, but not-yet deprecated or planned for removal.